src/julia-config-p.h.in

```diff
@@ -0,0 +1,9 @@
+//
+
+#ifndef JULIA_CONFIG_P_H
+#define JULIA_CONFIG_P_H
+
+#define JL_SYSTEM_IMAGE_PATH "@JL_SYSTEM_IMAGE_PATH@"
```

Comment:
The main reason I had this is actually I couldn't figure out how to properly escape the `"` when passing it through `CMAKE_C_FLAGS` (which was necessary when interfacing with the makefile system before all the c/cpp were ported). This could easily be changed later when we bring back the Makefile system, which I'm deleting on this branch right now because it is easier to see what targets I still need to port that way.